sdks/python/apache_beam/pipeline.py

```diff
@@ -1226,7 +1255,7 @@ def transform_to_runner_api(
     transform_urn = transform_spec.urn if transform_spec else None
     if (not environment_id and
         (transform_urn not in Pipeline.runner_implemented_transforms())):
-      environment_id = context.default_environment_id()
+      environment_id = context.get_environment_id_for_transform(self.transform)
```

Comment:
Let's just pass the hints here, rather than the transform itself (which makes it seem like the context cares about the type or other attributes of the transform).